### ISSUE TITLE
[Develop branch] Relative line-numbers does not work when using property list

### ISSUE THREAD

**Issue by am** [syl20bnr/spacemacs#11272]
#### Description :octocat:
When setting options for dotspacemacs-line-number the 'relative line-numbers don't show:

```lisp
dotspacemacs-line-numbers '(:relative nil
                            :disabled-for-modes dired-mode
                                                markdown-mode
                                                org-mode
                                                text-mode
                            :size-limit-kb 1000)
```
Setting the value without any customization works as expected:

```lisp
dotspacemacs-line-numbers 'relative
```

I've tried the same example found in documentation:
```lisp
dotspacemacs-line-numbers '(:relative nil
                            :enabled-for-modes prog-mode
                            :disabled-for-modes markdown-mode
                            :size-limit-kb 1000)
```
Am I missing something?

#### Reproduction guide :beetle:
- Start Emacs
- Open .spacemacs file

*Observed behaviour:* :eyes: :broken_heart:
Relative line numbers are not enabled.

*Expected behaviour:* :heart: :smile:
Relative line number should be enabled.

#### System Info :computer:
- OS: darwin
- Emacs: 26.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. e641e6c6e)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: ivy
- Layers:
```elisp
(ivy auto-completion emacs-lisp git github markdown neotree org spell-checking writeroom syntax-checking
     (version-control :variables version-control-diff-side 'left version-control-diff-tool 'diff-hl)
     javascript html themes-megapack theming colors ranger)
```
- System configuration features: NOTIFY ACL LIBXML2 ZLIB TOOLKIT_SCROLL_BARS THREADS

#### Backtrace :paw_prints:
```
[...]
Loading /Users/am/.emacs.d/layers/+spacemacs/spacemacs-defaults/keybindings.el (source)...done
Loading /Users/am/.emacs.d/layers/+source-control/version-control/keybindings.el (source)...done
Loading /Users/am/.emacs.d/.cache/spacemacs-buffer.el (source)...done
Starting new Ispell process /usr/local/bin/ispell with default dictionary...
Spacemacs is ready.
(Spacemacs) Calling dotfile user config...
Configuring package whitespace...done
Configuring package fill-column-indicator...done
(Spacemacs) Calling dotfile Emacs custom settings...
Configuring package recentf...done
Loading /Users/am/.emacs.d/.cache/recentf...done
Configuring package grep...done
Configuring package projectile...done
Skipping check for new version (reason: dotfile) [2 times]
Configuring package diff-mode...done
Configuring package vc...done
Configuring package diff-hl...done
Configuring package evil-escape...done
Configuring package window-purpose...done
Configuring package rainbow-mode...done
Configuring package yasnippet...done
[yas] Prepared just-in-time loading of snippets successfully.
Configuring package smartparens...done
Configuring package highlight-parentheses...done
Configuring package flycheck...done
Configuring package evil-lisp-state...done
Configuring package auto-highlight-symbol...done
Configuring package elisp-slime-nav...done
Configuring package auto-compile...done
Configuring package company...done
Configuring package evil-surround...done
```

**Comment by bmag** [comment]
1. works for me. In elisp files (e.g. `.spacemacs`) I get absolute line numbers, as expected

2. if you want relative numbers, you need `:relative t`, not `:relative nil`

3. it isn't clear if you are seeing absolute line numbers, or no line numbers at all

What kind (absolute/relative/none) of line numbers do you see in `.emacs.d/init.el` buffer? How about in other buffers? 99% chance the problem isn't with `dotspacemacs-line-numbers`

**Comment by am** [comment]
@bmag thank you for reviewing this. I though it was clear about my intentions:

> Observed behaviour: 👀 💔
> Relative line numbers are not enabled.

> Expected behaviour: ❤️ 😄
> Relative line number should be enabled.

**You are absolutely right**, my problem is that I was setting to `nil` instead of `t`. I was not able to understand the difference when reading the documentation. I thought that only by setting `:relative` it was it, and the `nil` was there for something else. [Reading documentation](https://github.com/syl20bnr/spacemacs/blob/develop/doc/DOCUMENTATION.org#global-line-numbers) once again and clearly differentiate `Relative line numbers` and `Enable line numbers`, each one with the following example, and I did miss the one that puts `:relative t`. 

Closing this since the problem is solved already. Thank you!
